astropy/io/misc/tests/test_hdf5.py

```diff
@@ -567,6 +567,23 @@ def test_read_h5py_objects(tmpdir):
     f.close()         # don't raise an error in 'test --open-files'
 
 
+@pytest.mark.skipif('not HAS_H5PY')
+def test_write_to_Hdf5(tmpdir):
+    test_file = str(tmpdir.join('test.hdf5'))
+    t1 = Table()
+    t1['x']=['a','b','c']
+    t1['y']=[ 1 , 2 , 3 ]
+    t1['z']=[b'a',b'b',b'c']
+    t1.write(test_file, path='the_table',overwrite=True)
+    t=Table.read(test_file, path='the_table')
+    assert np.all(t1['x']==['a','b','c'])
+    assert np.all(t1['y']==[ 1 , 2 , 3 ])
+    assert np.all(t1['z']==[b'a',b'b',b'c'])
+    assert np.all(t['x']==['a','b','c'])
+    assert np.all(t['y']==[ 1 , 2 , 3 ])
+    assert np.all(t['z']==[b'a',b'b',b'c'])
```

Comment:
I don't understand why this test does not fail, after writing the file the 'x' and 'z' column should behave the same way, i.e. should both be read as bytes and decoded to unicode with the unicode sandwich.

Reply:
i don't think it should fail-
>>> tx['x'],tx['y'] = ['1','2','3'],[b'1',b'2',b'3']
>>> tx['x']==tx['y']
[ True  True  True]

Reply:
@saimn should I remove this test?

Reply:
@ananyashreyjain - No, I was wrong ;), it also works.